### ISSUE TITLE
GetSchema("Columns") fails for LONGTEXT column

### ISSUE THREAD

**Issue by Rtaglia** [mysql-net/MySqlConnector#802]
Hi,

Trying to read the metadata of mysql, the function GetSchema() failed because of CHARACTER_MAXIMUM_LENGTH. 

DataTable columnsTable = connection.GetSchema("Columns", tableRestrictions);

I replace it by

```csharp
public DataTable getMySqlColumns(string schemaName, string tableName)
        {
            DataTable columnTable = new DataTable();
            MySqlDataAdapter adapter = new MySqlDataAdapter();
            adapter.SelectCommand = (MySqlCommand)connection.CreateCommand();
            adapter.SelectCommand.CommandText = $@"
SELECT 
	COLUMN_NAME, DATA_TYPE, IS_NULLABLE, CHARACTER_MAXIMUM_LENGTH 
FROM 
	information_schema.columns
WHERE
	TABLE_SCHEMA='{schemaName}' AND TABLE_NAME='{tableName}'
ORDER BY
    ORDINAL_POSITION ASC;
";
            adapter.Fill(columnTable);
            return columnTable;
        }
```

**Comment by bgrainger** [comment]
> Trying to read the metadata of mysql, the function GetSchema() failed because of CHARACTER_MAXIMUM_LENGTH.

Can you provide more details on what specifically failed? Was there an exception message and call stack?

What MySQL Server software and version are you using?

**Comment by Rtaglia** [comment]
Hello,

Here is the stacktrace :

System.ArgumentException: Value was either too large or too small for an Int32.Couldn't store <4294967295> in CHARACTER_MAXIMUM_LENGTH Column.  Expected type is Int32.
 ---> System.OverflowException: Value was either too large or too small for an Int32.
   at System.Convert.ThrowInt32OverflowException()
   at System.Convert.ToInt32(Int64 value)
   at System.Int64.System.IConvertible.ToInt32(IFormatProvider provider)
   at System.Data.Common.Int32Storage.Set(Int32 record, Object value)
   at System.Data.DataColumn.set_Item(Int32 record, Object value)
   --- End of inner exception stack trace ---
   at System.Data.DataColumn.set_Item(Int32 record, Object value)
   at System.Data.DataTable.NewRecordFromArray(Object[] value)
   at System.Data.DataRowCollection.Add(Object[] values)
   at MySqlConnector.Core.SchemaProvider.FillDataTable(DataTable dataTable, String tableName) in /_/src/MySqlConnector/Core/SchemaProvider.cs:line 984
   at MySqlConnector.Core.SchemaProvider.FillColumns(DataTable dataTable) in /_/src/MySqlConnector/Core/SchemaProvider.cs:line 193
   at MySqlConnector.Core.SchemaProvider.GetSchema(String collectionName) in /_/src/MySqlConnector/Core/SchemaProvider.cs:line 60
   at MySql.Data.MySqlClient.MySqlConnection.GetSchema(String collectionName, String[] restrictions) in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlConnection.cs:line 394


I'm using the last Mysql database server wich is 8.0.19

Thank you for your help



________________________________
De : Bradley Grainger <notifications@github.com>
Envoyé : samedi 18 avril 2020 20:05
À : mysql-net/MySqlConnector <MySqlConnector@noreply.github.com>
Cc : Rtaglia <roberto_taglia@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [mysql-net/MySqlConnector] GetSchema() failed (#802)


Trying to read the metadata of mysql, the function GetSchema() failed because of CHARACTER_MAXIMUM_LENGTH.

Can you provide more details on what specifically failed? Was there an exception message and call stack?

What MySQL Server software and version are you using?

**Comment by bgrainger** [comment]
Fixed in [0.64.1](https://github.com/mysql-net/MySqlConnector/releases/tag/0.64.1).
